src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -232,6 +233,42 @@ private void SetVerb()
                 }
             }
 
+            private void SetVersion()
+            {
+                Version v = _requestMessage.Version;
+                if (v != null)
+                {
+                    // Try to use the requested version, if a known version was explicitly requested.
+                    // If an unknown version was requested, we simply use libcurl's default.
+                    var curlVersion =
+                        (v.Major == 1 && v.Minor == 1) ? Interop.Http.CurlHttpVersion.CURL_HTTP_VERSION_1_1 :
+                        (v.Major == 1 && v.Minor == 0) ? Interop.Http.CurlHttpVersion.CURL_HTTP_VERSION_1_0 :
+                        (v.Major == 2 && v.Minor == 0) ? Interop.Http.CurlHttpVersion.CURL_HTTP_VERSION_2_0 :
+                        Interop.Http.CurlHttpVersion.CURL_HTTP_VERSION_NONE;
+
+                    if (curlVersion != Interop.Http.CurlHttpVersion.CURL_HTTP_VERSION_NONE)
+                    {
+                        // Ask libcurl to use the specified version if possible.
+                        CURLcode c = Interop.Http.EasySetOptionLong(_easyHandle, CURLoption.CURLOPT_HTTP_VERSION, (long)curlVersion);
```

Comment:
Good suggestion.  I added an additional commit for setting pipelining.  Note that the pipelining option is at the multi handle level rather than the easy handle level, so I've set it on all multi handles; libcurl should then apply it when using HTTP/2 for individual easy requests on that multi handle.